packages/runtime-common/realm.ts

```diff
@@ -254,6 +256,7 @@ export class Realm {
     ],
   ]);
   #assetsURL: URL;
+  #dbAdapter: DBAdapter;
```

Comment:
where do you actually start this adapter? it would seem that you are relying on the search index to start this adapter, as previously this adapter was only ever used in the search index. I think this is key to the issues that you have in the loader tests. Your start and end for this adapter need to be paired, like they are in the search index. It should be ok to try to start an adapter mulitple times. if it is already started nothing will happen. but cirtically you need to start it before you start using it. starting it is async, so probably you need to start it in Realm.#startup().

Reply:
or wait, i think i misspoke, the dbAdapter that is passed in should _already_ be started. so probably that is not the issue here